### PR TITLE
838 signer implement last low priority methods

### PR DESCRIPTION
**Pull request by lucanicoladebiasi**
# Description

* EIP-191 `signMessage` and and EIP-712 `signTypedData` implemented.
* `packages/core/src/utils/txt/txt.ts` introduces the `txt.encode` and `txt.decode` methods to enforce consistent text-binary codec.
* Imports from `noble*` packages are prefixed with `n_` to avoid namespace collisions.

# Fixes # (issue)

Text codec implemented enforcing a consistent `NFC` unicode normalisation for utf-8 binary representation.

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] `yarn test:solo`
- [x] `yarn test:unit`

**Test Configuration**:
* Node.js Version: v20.14.0
* Yarn Version: 1.22.22

# Checklist:

- [x] My code follows the coding standards of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] New and existing integration tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have not added any vulnerable dependencies to my code

### CODE REVIEW

File: packages/network/src/signer/signers/vechain-private-key-signer/vechain-private-key-signer.ts [hunk 1]
```diff
@@ -114,6 +124,78 @@ class VeChainPrivateKeySigner extends VeChainAbstractSigner {
...
+    async signTypedData(
+        domain: vechain_sdk_core_ethers.TypedDataDomain,
+        types: Record<string, vechain_sdk_core_ethers.TypedDataField[]>,
+        value: Record<string, unknown>
+    ): Promise<string> {
+        return await new Promise((resolve, reject) => {
+            try {
+                const hash = n_utils.hexToBytes(
+                    Hex.canon(
+                        ethers.TypedDataEncoder.hash(domain, types, value)
```
Comment (rodolfopietro97) [inline packages/network/src/signer/signers/vechain-private-key-signer/vechain-private-key-signer.ts:183]: HI @lucanicoladebiasi can we substitute ethers here in the future?

File: packages/network/tests/signer/signers/vechain-private-key-signer/fixture.ts [hunk 2]
```diff
@@ -14,6 +14,34 @@ import {
     TransactionDelegationError
 } from '@vechain/sdk-errors';
 
+interface TestCaseTypedDataDomain {
```
Comment (rodolfopietro97) [inline packages/network/tests/signer/signers/vechain-private-key-signer/fixture.ts:17]: Can you add more comments on types?

**Review by rodolfopietro97 — CHANGES_REQUESTED** [review]
@lucanicoladebiasi Great work.
Only 2 few things:
* 2 feedback (1 is only a question)
* And other one is why implementing and change some crypto stuff in a pr related to signer low priority method. BUT no problem on that, it is only a question

File: packages/network/src/signer/signers/vechain-private-key-signer/vechain-private-key-signer.ts (hunk 1 above)
Comment (lucanicoladebiasi) [reply packages/network/src/signer/signers/vechain-private-key-signer/vechain-private-key-signer.ts:183]: Yes, and you know I'm a strong advocate to do not use `ethers`, but for this function this is not possible now. The `signTypeData` method leverages on ABI codec, we haven't our implementation yet, hence until we do not provide our oroginal ABI codec doesn't make sense to do the same `ethers.TypedDataEncoder.hash(domain, types, value)`.

File: packages/network/tests/signer/signers/vechain-private-key-signer/fixture.ts (hunk 2 above)
Comment (rodolfopietro97) [reply packages/network/tests/signer/signers/vechain-private-key-signer/fixture.ts:22]: https://github.com/vechain/vechain-sdk-js/pull/1004/commits/c7bd734f09ceed21b3a996a542a2f371d375350c

File: packages/network/src/signer/signers/vechain-private-key-signer/vechain-private-key-signer.ts (hunk 1 above)
Comment (rodolfopietro97) [reply packages/network/src/signer/signers/vechain-private-key-signer/vechain-private-key-signer.ts:183]: lgtm

**Review by rodolfopietro97 — APPROVED** [review]

**Pull request merged**